### PR TITLE
feat: port Data.Nat.Bits

### PR DESCRIPTION
**Pull request by Shreyas4991**
First PR for `data.nat.bits`. Copied file Data/Nat/Bits.lean from mathlib3port as is. 
Mathlib commit : d012cd09a9b256d870751284dd6a29882b0be105

### CODE REVIEW

**Comment by semorrison** [comment]
Please reopen this as a PR from a branch on this repository, so that CI can run.

File: mathlib3port_commit.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+d012cd09a9b256d870751284dd6a29882b0be105
```
Comment (rwbarton) [inline mathlib3port_commit.txt:1]: I don't think this file was intended to be committed.

Comment (Shreyas4991) [reply mathlib3port_commit.txt:1]: Yeah sorry. I shall remove it

**Comment by Shreyas4991** [comment]
Hi @semorrison 

Is there a way to move this PR as required?

**Pull request merged**